Lobby/EOSLobby.cs

```diff
@@ -0,0 +1,197 @@
+            if (lobbyData != null)
+            {
+                foreach (AttributeData data in lobbyData)
+                {
+                    modHandle.AddAttribute(new LobbyModificationAddAttributeOptions { Attribute = data, Visibility = LobbyAttributeVisibility.Public });
+                    lobbyReturnData.Add(new Attribute { Data = data, Visibility = LobbyAttributeVisibility.Public });
+                }
+            }
+
+            //update the lobby
+            EOSSDKComponent.GetLobbyInterface().UpdateLobby(new UpdateLobbyOptions { LobbyModificationHandle = modHandle }, null, (UpdateLobbyCallbackInfo _) => { });
+
+            isLobbyOwner = true;
+            ConnectedToLobby = true;
+            EOSSDKComponent.GetLobbyInterface().CopyLobbyDetailsHandle(new CopyLobbyDetailsHandleOptions { LobbyId = callback.LobbyId, LocalUserId = EOSSDKComponent.LocalUserProductId }, out ConnectedLobbyDetails);
+            currentLobbyId = callback.LobbyId;
+
+            //invoke event
+            JoinLobbyComplete.Invoke(lobbyReturnData);
+        });
+    }
+
+    //find lobbies
+    public virtual void FindLobbies(uint maxResults, LobbySearchSetParameterOptions[] lobbySearchSetParameterOptions)
+    {
+        //create search handle and list of lobby details
+        LobbySearch search = new LobbySearch();
+
+        //set the search handle
+        EOSSDKComponent.GetLobbyInterface().CreateLobbySearch(new CreateLobbySearchOptions { MaxResults =  maxResults}, out search);
+
+        //set search parameters
+        foreach (LobbySearchSetParameterOptions searchOption in lobbySearchSetParameterOptions)
+        {
+            search.SetParameter(searchOption);
+        }
+        
+        //find lobbies
+        search.Find(new LobbySearchFindOptions { LocalUserId = EOSSDKComponent.LocalUserProductId }, null, (LobbySearchFindCallbackInfo callback) => 
+        {
+            List<LobbyDetails> foundLobbies = new List<LobbyDetails>();
+
+            //if the search was unsuccessful, log an error and return
+            if (callback.ResultCode != Result.Success)
+            {
+                Debug.LogError("There was an error while finding lobbies. Error: " + callback.ResultCode);
+                return;
+            }
+
+            foundLobbies.Clear();
+
+            //for each lobby found, add data to details
+            for (int i = 0; i < search.GetSearchResultCount(new LobbySearchGetSearchResultCountOptions { }); i++)
+            {
+                LobbyDetails lobbyInformation;
+                search.CopySearchResultByIndex(new LobbySearchCopySearchResultByIndexOptions { LobbyIndex = (uint)i }, out lobbyInformation);
+                foundLobbies.Add(lobbyInformation);
+            }
+
+            //invoke event
+            FindLobbiesComplete.Invoke(foundLobbies);
+        });
+    }
+
+    //join lobby
+    public virtual void JoinLobby(LobbyDetails lobbyToJoin, string[] attributeKeys, bool presenceEnabled = false)
```

Comment:
Probably should have default values too and we should have a second method that lets users join with the lobby id string too

Reply:
How would you join a lobby via ID?

Reply:
You can do that, but I thinking about it again probably not a common use case